src/extension/op_def.rs

```diff
@@ -70,28 +70,47 @@ pub trait CustomLowerFunc: Send + Sync {
     ) -> Option<Hugr>;
 }
 
+/// Type parameters passed to [func]. (The returned [PolyFuncType]
+/// may require further type parameters, not declared here.)
+pub struct CustomFunc {
+    static_params: Vec<TypeParam>,
```

Comment:
Now you've done your #678, I'd say moving this static-params getter into `trait CustomSignatureFunc` might be a good next step (perhaps before the rest of this?)

Reply:
Done in #706